server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1347,7 +1345,8 @@ public static MetaData fromXContent(XContentParser parser, boolean preserveUnkno
                 currentFieldName = parser.currentName();
             }
 
-            if (!"meta-data".equals(parser.currentName())) {
+            if (!"meta-data".equals(parser.currentName()) &&
+                !"metadata".equals(parser.currentName())) {
```

Comment:
As commented above, let's remove the lenient parsing here.

Reply:
Done. This also caused some test cases to be broken and removed as they relied on deserialization of API-serialized XContent https://github.com/elastic/elasticsearch/pull/52743/commits/0c11613501e979d9202a1f6fa7dda3aca4b69b72